src/crimson/osd/main.cc

```diff
@@ -79,26 +122,36 @@ int main(int argc, char* argv[])
   }
   std::string cluster_name;
   std::string conf_file_list;
+
   // ceph_argparse_early_args() could _exit(), while local_conf() won't ready
   // until it's started. so do the boilerplate-settings parsing here.
   auto init_params = ceph_argparse_early_args(ceph_args,
                                               CEPH_ENTITY_TYPE_OSD,
                                               &cluster_name,
                                               &conf_file_list);
+
+
   seastar::sharded<OSD> osd;
   seastar::sharded<ceph::net::SocketMessenger> cluster_msgr, client_msgr;
   seastar::sharded<ceph::net::SocketMessenger> hb_front_msgr, hb_back_msgr;
   using ceph::common::sharded_conf;
   using ceph::common::sharded_perf_coll;
   using ceph::common::local_conf;
+  ObserverMgr<ceph::md_config_obs_impl<ConfigProxy>> config_tracker;
+  ConfigValues config_values;
```

Comment:
i'd suggest use a `unique_ptr<ConfigValues>`. because `ConfigValue` has relatively large memory foot print, so

- there is no need to keep it around after passing it to `sharded_conf().start()`.
- we can **move** it to sharded_conf, so no memory copy when constructing the shard.0 of sharded_conf.